src/main/java/edu/harvard/iq/dataverse/engine/command/impl/CreateDatasetCommand.java

```diff
@@ -123,28 +124,34 @@ public Dataset execute(CommandContext ctxt) throws CommandException {
         dsv.setCreateTime(createDate);
         dsv.setLastUpdateTime(createDate);
         theDataset.setModificationTime(createDate);
-        for (DataFile dataFile: theDataset.getFiles() ){
-            dataFile.setCreator((AuthenticatedUser)  getRequest().getUser());
+        for (DataFile dataFile : theDataset.getFiles()) {
+            dataFile.setOwner(theDataset);
+            dataFile.setCreator((AuthenticatedUser) getRequest().getUser());
             dataFile.setCreateDate(theDataset.getCreateDate());
         }
         String nonNullDefaultIfKeyNotFound = "";
-        String    protocol = ctxt.settings().getValueForKey(SettingsServiceBean.Key.Protocol, nonNullDefaultIfKeyNotFound);
-        String    authority = ctxt.settings().getValueForKey(SettingsServiceBean.Key.Authority, nonNullDefaultIfKeyNotFound);
-        String  doiSeparator = ctxt.settings().getValueForKey(SettingsServiceBean.Key.DoiSeparator, nonNullDefaultIfKeyNotFound);
-        String    doiProvider = ctxt.settings().getValueForKey(SettingsServiceBean.Key.DoiProvider, nonNullDefaultIfKeyNotFound);
-        if (theDataset.getProtocol()==null) theDataset.setProtocol(protocol);
-        if (theDataset.getAuthority()==null) theDataset.setAuthority(authority);
-        if (theDataset.getDoiSeparator()==null) theDataset.setDoiSeparator(doiSeparator);
-        if(theDataset.getStorageIdentifier()==null) {
+        String protocol = ctxt.settings().getValueForKey(SettingsServiceBean.Key.Protocol, nonNullDefaultIfKeyNotFound);
+        String authority = ctxt.settings().getValueForKey(SettingsServiceBean.Key.Authority, nonNullDefaultIfKeyNotFound);
+        String doiSeparator = ctxt.settings().getValueForKey(SettingsServiceBean.Key.DoiSeparator, nonNullDefaultIfKeyNotFound);
+        String doiProvider = ctxt.settings().getValueForKey(SettingsServiceBean.Key.DoiProvider, nonNullDefaultIfKeyNotFound);
+        if (theDataset.getProtocol() == null) {
+            theDataset.setProtocol(protocol);
+        }
+        if (theDataset.getAuthority() == null) {
+            theDataset.setAuthority(authority);
+        }
+        if (theDataset.getDoiSeparator() == null) {
+            theDataset.setDoiSeparator(doiSeparator);
+        }
+        if (theDataset.getStorageIdentifier() == null) {
             //FIXME: if the driver identifier is not set in the JVM options, should the storage identifier be set to file b default, or should an exception be thrown?
```

Comment:
(asking, as I'm unfamiliar with this area) - is this fixed? Can it be fixed as part of this PR?

Reply:
In my opinion should be set to file, but @landreev should probably confirm.

Reply:
Yes, defaulting to "file://" is correct. 
It's not a "FIXME", strictly speaking. More like "confirm that ..." Also, note that this comment has been there for a while (as in, it's not being introduced in this branch). I don't think it should be holding this PR either.